PythonVisualizations/PriorityQueue.py

```diff
@@ -0,0 +1,367 @@
+        return add_vector(self.cellCoords(index), (half_cell, half_cell))
+
+    def createArrayCell(self, index):  # Create a box representing an array cell
+        cell_coords = self.cellCoords(index)
+        half_border = self.CELL_BORDER // 2
+        rect = self.canvas.create_rectangle(
+            *add_vector(cell_coords,
+                        (-half_border, -half_border,
+                         self.CELL_BORDER - half_border, self.CELL_BORDER - half_border)),
+            fill=None, outline=self.CELL_BORDER_COLOR, width=self.CELL_BORDER)
+        self.canvas.lower(rect)
+        return rect
+
+    def createCellValue(self, indexOrCoords, key, color=None):
+        """Create new canvas items to represent a cell value.  A square
+        is created filled with a particular color with an text key centered
+        inside.  The position of the cell can either be an integer index in
+        the Array or the bounding box coordinates of the square.  If color
+        is not supplied, the next color in the palette is used.
+        An event handler is set up to update the VisualizationApp argument
+        with the cell's value if clicked with any button.
+        Returns the tuple, (square, text), of canvas items
+        """
+        # Determine position and color of cell
+        if isinstance(indexOrCoords, int):
+            rectPos = self.cellCoords(indexOrCoords)
+            valPos = self.cellCenter(indexOrCoords)
+        else:
+            rectPos = indexOrCoords
+            valPos = divide_vector(add_vector(rectPos[:2], rectPos[2:]), 2)
+        if color is None:
+            # Take the next color from the palette
+            color = drawable.palette[self.nextColor]
+            self.nextColor = (self.nextColor + 1) % len(drawable.palette)
+
+        cell_rect = self.canvas.create_rectangle(
+            *rectPos, fill=color, outline='', width=0)
+        cell_val = self.canvas.create_text(
+            *valPos, text=str(key), font=self.VALUE_FONT, fill=self.VALUE_COLOR)
+        handler = lambda e: self.setArgument(str(key))
+        for item in (cell_rect, cell_val):
+            self.canvas.tag_bind(item, '<Button>', handler)
+
+        return cell_rect, cell_val
+
+    def display(self):
+        callEnviron = self.createCallEnvironment()
+        self.canvas.delete("all")
+
+        for i in range(self.size):  # Draw grid of cells
+            self.createArrayCell(i)
+
+        # go through each Drawable in the list
+        for i, n in enumerate(self.list):
+            if i and n:
+                # create display objects for the associated Drawables
+                n.display_shape, n.display_val = self.createCellValue(
+                    i, n.val, n.color)
+                n.color = self.canvas.itemconfigure(n.display_shape, 'fill')
+
+        #self.index = self.createIndex(self.nItems-1, 'front', level=-1) # indicate priority
+        callEnviron |= set(self.index)
```

Comment:
It looks like you tried creating the `nItems` index at this point.  You had the right idea.  Change lines 241-242 to:
```
        self.index = self.createIndex(self.nItems, 'nItems', level=-1)
        #callEnviron |= set(self.index)
```
* I've changed the label to `nItems` because we call it `__nItems` in the textbook code
* It's initialized to the value of the internal `self.nItems` and not offset by 1.
* It should **not** be put in the `callEnviron` since it stays on the canvas over all operations.  It's a field of the PriorityQueue data structure.  
* You should remove lines 242 and 227.  You don't need to create a call environment in this `display()` method because it's not an operational call that we're animating.  It's not going to push a call like `insert()`, `remove()`, or `peek()` on the call stack.